pkg/operator/controller/gatewayclass/subscription.go

```diff
@@ -115,3 +121,76 @@ func subscriptionChanged(current, expected *operatorsv1alpha1.Subscription) (boo
 
 	return true, updated
 }
+
+// ensureServiceMeshOperatorInstallPlan attempts to ensure that the install plan for the appropriate OSSM operator
+// version is approved.
+func (r *reconciler) ensureServiceMeshOperatorInstallPlan(ctx context.Context) (bool, *operatorsv1alpha1.InstallPlan, error) {
+	haveInstallPlan, current, err := r.currentInstallPlan(ctx)
+	if err != nil {
+		return false, nil, err
+	}
+	switch {
+	case !haveInstallPlan:
+		// The OLM operator creates the initial InstallPlan, so if it doesn't exist yet, do nothing.
+		return false, nil, nil
+	case haveInstallPlan:
+		desired, err := desiredInstallPlan(current)
+		if err != nil {
+			return true, current, err
```

Comment:
At the moment, `desiredInstallPlan` only returns a nil error.  If that doesn't change, it's probably fine to remove the returned err and don't check it.